resample/empirical.py

```diff
@@ -45,15 +47,12 @@ def quantile_fn(sample: np.ndarray) -> Callable:
     """
     sample = np.sort(sample)
     n = len(sample)
-    prob = np.arange(1, n + 1, dtype=float) / n
 
     def quant(p):
         if not 0 <= p <= 1:
             raise ValueError("Argument must be between zero and one")
-        if p < 1 / n:
-            # TODO: How to handle this case?
-            return sample[0]
-        return _interp1d(prob, sample)(p)
+        idx = max(ceil(p * n) - 1, 0)
+        return sample[idx]
```

Comment:
The implementation should be vectorized.
```py
q = quantile_fn([1, 2, 3])
q([1, 2, 3]) # this should work for any array-like
```

Reply:
I added a vectorized implementation. The implementation returns `nan` for invalid probabilities, like `scipy.stats.[...].ppf`

Reply:
Can you ask before redoing the implementation of a PR? Vectorizing is good but I still think it should return an error when we go outside [0, 1]. This is how it's done in both R and pandas, and like we're saying scipy sometimes makes unusual choices.

Reply:
Ok, sorry. Should I revert this?

I don't agree that raising an error makes more sense. It makes less sense. It makes no sense to fail a whole array operation if only one of the probabilities is invalid. The scipy approach is more sensible, it returns an "error state" per result instead of making the complete operation fail.

Reply:
Seems like pandas and R are making unusual choices.

Reply:
An array with an invalid entry is an invalid array, so I don't see why it's wrong to return an error. Seems pretty reasonable to expect a program to fail loudly that passes a negative number where a probability is required.

In any case it's not a huge deal so merging essentially as-is (I made a change to return a scalar instead of a singleton array for scalar input, which I think makes sense).